src/pamela/parser.clj

```diff
@@ -1249,8 +1249,10 @@
                          ;; _ (log/warn "PRE-VALIDATE")
                          pir (if check-only?
                                {:tree tree0}
-                               (validate-pamela
-                                 (insta/transform pamela-ir tree0)))]
+                               (if do-not-validate
+                                 (insta/transform pamela-ir tree0)
+                                 (validate-pamela
+                                   (insta/transform pamela-ir tree0))))]
```

Comment:
A simple improvement, that eliminates the redundancy:
(let [validation-fn (if do-not-validate identity validate-pamela)]
 (validation-fn (insta/transform pamela-ir tree0))